server/task/task-distributed/src/main/scala/org/apache/james/task/eventsourcing/distributed/TaskEventDTO.scala

```diff
@@ -74,18 +77,25 @@ object CancelRequestedDTO {
 case class CompletedDTO(@JsonProperty("type") typeName: String,
                         @JsonProperty("aggregate") aggregateId: String,
                         @JsonProperty("event") eventId: Int,
-                        @JsonProperty("result") getResult: String)
+                        @JsonProperty("result") getResult: String,
+                        @JsonProperty("additionalInformation") getAdditionalInformation: Optional[String])
   extends TaskEventDTO(typeName, aggregateId, eventId) {
-  def toDomainObject: Completed = Completed(domainAggregateId, domainEventId, domainResult)
+  def toDomainObject(jsonTaskAdditionalInformationsSerializer: JsonTaskAdditionalInformationsSerializer): Completed = {
+    val deserializedAdditionalInformation = getAdditionalInformation.asScala.map(jsonTaskAdditionalInformationsSerializer.deserialize)
```

Comment:
why it's the responsibility of the DTO to build a sub-DTO based on a json string and a deserializer?
Why it's not done before calling `toDomainObject`?

Reply:
Will be seen in a next PR